frontend/src/components/parties/PartyProfile.js

```diff
@@ -36,112 +37,108 @@ const propTypes = {
-  mineBasicInfoList: PropTypes.arrayOf(CustomPropTypes.mine),
-  match: PropTypes.object,
+  partyRelationshipTypeHash: PropTypes.objectOf(PropTypes.strings),
+  mineBasicInfoListHash: PropTypes.objectOf(PropTypes.strings),
+  match: CustomPropTypes.match.isRequired,
 };
 
 const defaultProps = {
   partyRelationships: [],
-  partyRelationshipTypes: [],
-  mineBasicInfoList: [],
+  partyRelationshipTypeHash: {},
+  mineBasicInfoListHash: {},
 };
 
 export class PartyProfile extends Component {
+  state = { isLoaded: false };
+
   componentDidMount() {
     const { id } = this.props.match.params;
     this.props.fetchPartyById(id);
     this.props.fetchPartyRelationships({ party_guid: id }).then(() => {
       const mine_guids = uniq(this.props.partyRelationships.map(({ mine_guid }) => mine_guid));
-      this.props.fetchMineBasicInfoList([...mine_guids]);
+      this.props.fetchMineBasicInfoList([...mine_guids]).then(() => {
+        this.setState({ isLoaded: true });
+      });
     });
     this.props.fetchPartyRelationshipTypes();
   }
 
-  getpartyRelationshipTitle(partyRelationship) {
-    const partyRelationshipType = this.props.partyRelationshipTypes.find(
-      ({ value }) => value === partyRelationship.mine_party_appt_type_code
-    );
-    return (partyRelationshipType && partyRelationshipType.label) || String.EMPTY;
-  }
-
-  getMineName(mineId) {
-    const mine =
-      this.props.mineBasicInfoList.length > 0 &&
-      this.props.mineBasicInfoList.find(({ guid }) => guid === mineId);
-    return (mine && mine.mine_name) || String.LOADING;
-  }
-
   render() {
     const { id } = this.props.match.params;
     const parties = this.props.parties[id];
+    const columns = [
+      {
+        title: "Mine Name",
+        dataIndex: "mineName",
+        render: (text, record) => (
+          <div title="Mine Name">
+            <Link to={router.MINE_SUMMARY.dynamicRoute(record.mineGuid, "contacts")}>{text}</Link>
+          </div>
+        ),
+      },
+      {
+        title: "Role",
+        dataIndex: "role",
+        render: (text) => <div title="Role">{text}</div>,
+      },
+      {
+        title: "Dates",
+        dataIndex: "dates",
+        render: (text, record) => (
+          <div title="Dates">
+            {record.startDate} - {record.endDate}
+          </div>
+        ),
+      },
+    ];
 
-    const isLoaded =
-      this.props.partyRelationshipTypes.length > 0 &&
-      (this.props.partyRelationships.length > 0 && parties);
+    const transformRowData = (partyRelationships) =>
+      partyRelationships.map((relationship) => ({
+        key: relationship.mine_party_appt_guid,
+        mineGuid: relationship.mine_guid,
+        mineName: this.props.mineBasicInfoListHash[relationship.mine_guid],
+        role: this.props.partyRelationshipTypeHash[relationship.mine_party_appt_type_code],
+        endDate:
+          relationship.end_date === "9999-12-31" ? "Present" : formatDate(relationship.end_date),
```

Comment:
James recently merged a change that sets Mine Manager `mine_party_appt`s to `null` in the ETL process instead of to this bunk value. We should probably handle both cases here, to be safe.